engine/src/lib.rs

```diff
@@ -991,6 +1139,32 @@ mod contract {
         near_account_to_evm_address(predecessor_account_id.as_bytes())
     }
 
+    fn update_hashchain(io: &mut Runtime, method_name: &str, input: &[u8], output: &[u8]) {
+        let block_height = io.block_height();
+
+        let mut blockchain_hashchain = hashchain::get_state(io).unwrap_or_else(|_| {
```

Comment:
This reads the state of the hashchain or create a new one if the state does not exists.

I don't think this behave should go to production. We should ensure that the upgrade, or new etc. methods, create the state so it exists at this point.

Reply:
Yes, I agree. When we have a proper migration story in place then we should assume the state is present.

Reply:
Current behave is that Hashchain is off by default. It is needed to call `start_hashchain` method to start the hashchain.